.dotnet/src/Custom/Chat/ChatClient.Protocol.cs

```diff
@@ -26,4 +26,24 @@ public virtual async Task<ClientResult> CompleteChatAsync(BinaryContent content,
         PipelineResponse response = await Pipeline.ProcessMessageAsync(message, options).ConfigureAwait(false);
         return ClientResult.FromResponse(response);
     }
+
+    /// <inheritdoc cref="Internal.Chat.CreateChatCompletion(BinaryContent, RequestOptions)"/>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public virtual ClientResult CompleteChatStreaming(BinaryContent content, RequestOptions options = null)
```

Comment:
It might make sense to ACK with @KrzysztofCwalina that this is the intent for protocol methods in the unbranded client -- from [the gist he had shared last week ](https://gist.github.com/KrzysztofCwalina/c1297dca339ca338269594b571bc4bc9) I had understood he was thinking we'd address this with the "withers" pattern, but maybe that was only in the context of the Azure client?

Reply:
@annelo-msft, don't we have a way to turn buffering off using requestoptions?

Reply:
Today that is only available on message, but we could look at adding it to request options.

Reply:
I would rather add it on requestoptions, so we don't have to add methods like Travis is adding above.

Reply:
Sounds good.  I can create a PR for it relatively quickly.  I want to do a little work to remember if there are complications around that.

Reply:
Working on it now, but it just occurred to me that there is a flip-side question.  If we have convenience overloads called CompleteChatStreaming, are we OK if there is no corresponding protocol method?

i.e. we understand the goal of protocol methods here as being "one per endpoint" and can do everything the endpoint can do given any input supported?  And it's ok if some convenience methods have different names, particularly in the case that they have different return-types?

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/43909

Reply:
@trrwilson, I have a PR up for this now -- let's chat to understand how it would be easiest for you to consume this change from ClientModel.

Reply:
Merged this just now so tomorrow's dev build should have it, but LMK if you'd like it sooner and can totally accommodate that.